packages/keyring-controller/src/KeyringController.ts

```diff
@@ -626,6 +649,7 @@ export class KeyringController extends BaseController<
         vault: { persist: true, anonymous: false },
         isUnlocked: { persist: false, anonymous: true },
         keyrings: { persist: false, anonymous: false },
+        keyringsMetadata: { persist: false, anonymous: false },
```

Comment:
If we consider the ID and name of keyrings to be OK to be persisted outside of the vault, we could persist them only here in the state, so that:
1. We don't need to change how the Vault is updated, since metadata will not go there
2. We can remove `this.#keyringsMetadata`, as it would be redundant